hack/heap/hh_shared.c

```diff
@@ -104,6 +104,24 @@
 #include <sys/time.h>
 #include <sys/types.h>
 #include <unistd.h>
+
+#ifdef HAVE_MEMFD_CREATE
+#include <sys/memfd.h>
+#else
+#if HAVE_DECL___NR_MEMFD_CREATE
+#define HAVE_MEMFD_CREATE 1
+#include <asm/unistd.h>
+static int memfd_create(const char *name, unsigned int flags)
+{
```

Comment:
On Fri, Nov 06, 2015 at 02:26:43PM -0800, Gabe Levi wrote:
> > @@ -104,6 +104,24 @@
> >  #include <sys/time.h>
> >  #include <sys/types.h>
> >  #include <unistd.h>
> > +
> > +#ifdef HAVE_MEMFD_CREATE
> > +#include <sys/memfd.h>
> > +#else
> > +#if HAVE_DECL___NR_MEMFD_CREATE
> > +#define HAVE_MEMFD_CREATE 1
> > +#include <asm/unistd.h>
> > +static int memfd_create(const char *name, unsigned int flags)
> > +{
> 
> @int3 
> 
> > Nit: { should be on the same line as the rest of the declaration
> 
> > Also, can we have a comment here about what memfd_create does? As far as I understand it's the same as an atomic shm_open + mmap + unlink, right?

Ok, I will add a comment.

That's more like `shm_open + unlink` without the inconvenient of being limited by the size of `/dev/shm`. The file descriptor is not bound to a mount point.

The `mmap` has still to be made.